packages/abstractions/src/apiClientProxifier.ts

```diff
@@ -124,6 +124,9 @@ export const apiClientProxifier = <T extends object>(requestAdapter: RequestAdap
 					return apiClientProxifier(requestAdapter, getPathParameters(rawUrl), navigationMetadata, requestsMetadata);
 				};
 			}
+      if (name === "then") {
```

Comment:
It'd probably be safer to move this block before the error is thrown.
My worry is that if somebody has an API segment called "then" it will derail them.
At this other location, we'll know we didn't find any navigation metadata which makes for a more specific condition.
What do you think?